File: iou-cordapp/src/main/java/com/formssihk/sample/flow/CreateFlow.java
package com.formssihk.sample.flow;

import co.paralleluniverse.fibers.Suspendable;
import com.formssihk.sample.contract.IOUContract;
import com.formssihk.sample.state.IOUState;
import com.formssihk.sample.utils.CordaUtils;
import net.corda.core.contracts.UniqueIdentifier;
import net.corda.core.flows.FlowException;
import net.corda.core.flows.StartableByRPC;
import net.corda.core.transactions.SignedTransaction;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.time.Instant;
import java.util.HashMap;
import java.util.Map;

@StartableByRPC
public class CreateFlow extends IOUFlow {
    Logger log = LoggerFactory.getLogger(CreateFlow.class);

    String userPublicKey;
    Instant creationTime;
    Map<String, Instant> loanRecords = new HashMap<>();

    public CreateFlow(String userPublicKey, Instant creationTime,
                      Map<String, Instant> loanRecords) {
        super();
        this.userPublicKey = userPublicKey;
        this.creationTime = creationTime;
        this.loanRecords = loanRecords;
        command = new IOUContract.Commands.Create();
    }

    @Override
    @Suspendable
    public SignedTransaction call() throws FlowException {
        IOUState newState = new IOUState(
                userPublicKey,
                loanRecords,
                creationTime,
                Instant.now(),
                CordaUtils.getServiceHubAllParties(getServiceHub()),
                new UniqueIdentifier()
        );
        return defaultFlow(newState, null);
    }
}

File: iou-cordapp/src/main/kotlin/com/formssihk/sample/flow/CreateFlow.kt
package com.formssihk.sample.flow

import co.paralleluniverse.fibers.Suspendable
import com.formssihk.sample.contract.IOUContract
import com.formssihk.sample.state.IOUState
import com.formssihk.sample.utils.allParties
import com.formssihk.sample.utils.myIdentity
import net.corda.core.flows.StartableByRPC
import net.corda.core.transactions.SignedTransaction
import net.corda.core.utilities.loggerFor
import org.slf4j.Logger
import java.time.Instant

@StartableByRPC
class CreateFlow(val userPublicKey: String, val creationTime: Instant, val loanRecords: Map<String, Instant>) : IOUFlow(){
    override val log: Logger = loggerFor<CreateFlow>()
    override  val command = IOUContract.Commands.Create()

    @Suspendable
    override fun call(): SignedTransaction {
        val newState = IOUState(
            userPublicKey = userPublicKey,
            loadRecords = loanRecords,
            creationTime = creationTime,
            lastModifiedTime = Instant.now(),
            parties = serviceHub.allParties()
        )

        return defaultFlow(newState,null)
    }
}